### PR TITLE
Upgrade Dropwizard to 1.1.4, and other packages

### PR DESCRIPTION
**Pull request by zachschultz**
Pointing this to the most recent parent-pom, and upgrading dropwizard to 1.1.4

The extra exclusions/dependencies were needed because mvn test was throwing Dependency convergence errors, as it looks like there were some duplicate imports between the internal Dropwizard packages used

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,11 +5,11 @@
   <parent>
     <groupId>com.mainstreethub</groupId>
     <artifactId>parent-pom</artifactId>
-    <version>1.5.1</version>
+    <version>1.6.1-SNAPSHOT</version>
```
Comment (alexwen) [inline pom.xml]: This should be the release version, `1.6.0`

File: pom.xml [hunk 2]
```diff
@@ -72,21 +72,88 @@
...
   <dependencies>
     <!-- Dropwizard -->
+    <!-- The exclusions are required as Dropwizard seems to be having issues regarding the versioning
+     of packages its internal components use -->
     <dependency>
       <groupId>io.dropwizard</groupId>
       <artifactId>dropwizard-core</artifactId>
       <version>${dropwizard.version}</version>
+      <exclusions>
+        <exclusion>
```
Comment (alexwen) [inline pom.xml]: For these - where were the conflicts originating? Can you post the conflict output?

**Comment by zachschultz** [comment]
Here is what the dependency convergence issues look like. I only show guava and slf4j, but they occur for each dependency that I excluded and manually versioned.

image

File: pom.xml (hunk 2 above)
Comment (zachschultz) [reply pom.xml]: Just posted a screencap as a new comment

Comment (alexwen) [reply pom.xml]: I figured out how to simplify this a bit, if we pull in the dropwizard-bom:
```
  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>io.dropwizard</groupId>
        <artifactId>dropwizard-bom</artifactId>
        <version>${dropwizard.version}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>
```
And remove all specified versions, and exclusions, Dropwizard has already set everything up for us to not get conflicts.

File: pom.xml [hunk 3]
```diff
@@ -5,11 +5,11 @@
   <parent>
     <groupId>com.mainstreethub</groupId>
     <artifactId>parent-pom</artifactId>
-    <version>1.5.1</version>
+    <version>1.6.0</version>
   </parent>
 
   <artifactId>dropwizard-api-key-bundle</artifactId>
-  <version>1.0.1-SNAPSHOT</version>  <!-- Make sure to keep this in sync with the dropwizard version below. -->
+  <version>1.1.4-SNAPSHOT</version>  <!-- Make sure to keep this in sync with the dropwizard version below. -->
```
Comment (alexwen) [inline pom.xml]: We will likely want to set this up to have our new versioning scheme, where we have a build release number:
`1.1.4-1-SNAPSHOT`
